src/scheduler/server_info.c

```diff
@@ -274,6 +274,11 @@ query_server(status *pol, int pbs_sd)
 		pbs_statfree(bs_resvs);
 		return NULL;
 	}
+	if (!dflt_sched && sinfo->partition != NULL) {
+		sinfo->nodes = node_filter(sinfo->nodes, sinfo->num_nodes,
+					   node_partition_cmp, (void *) sinfo->partition, 0);
+		sinfo->num_nodes = count_array(sinfo->nodes);
```

Comment:
Thanks for finding it out, that was a big leak. I realized that I don't have to filter nodes because query_nodes already call node_in_partition function and dumps nodes that are not part of the partition.